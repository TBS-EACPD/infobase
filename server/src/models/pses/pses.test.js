const question_index_query = `
  query {
    root(lang:"en"){
      pses_questions{
        id
        name
        
        positive
        neutral
        negative
        
        agree
        type
        
        
      }

    }
  }
`

const org_query = `
  query($dept_code: String) {
    root(lang:"en"){
      org(dept_code:$dept_code) {
        pses_data {
          year
          
          answer1
          answer2
          answer3
          answer4
          answer5
          answer6
          answer7
          
          positive
          neutral
          negative
          
          agree
          
          
          average_1_to_5
          average_percent
          answer_count
          question {
            id
          }
          
        }
      }
    }
  }
`

const { execQuery } = global;

describe("PSES data", function(){

  it("has an index of all questions", async ()=> {
    const data = await execQuery(question_index_query);
    return expect(data).toMatchSnapshot();
  });

  it("has survey data for an org", async ()=> {
    const data = await execQuery(org_query, {dept_code: "AGR"});
    return expect(data).toMatchSnapshot();
  });


})