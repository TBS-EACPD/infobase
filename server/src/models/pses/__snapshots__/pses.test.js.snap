// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PSES data has an index of all questions 1`] = `
Object {
  "data": Object {
    "root": Object {
      "pses_questions": Array [
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q01",
          "name": "My department or agency implements activities and practices
that support a diverse workplace.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q02",
          "name": "Overall, my department or agency treats me with respect.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q03",
          "name": "I am encouraged to be innovative or to take initiative in my work.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q04",
          "name": "I believe I would be supported if I proposed a new idea, even though it might not work.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q05",
          "name": "I have support at work to balance my work and personal life.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q06",
          "name": "I am satisfied with my department or agency.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q07",
          "name": "Overall, I like my job.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q08",
          "name": "My immediate supervisor creates an environment where I feel free to discuss with him or her matters that affect my well-being at work.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q09",
          "name": "My immediate supervisor seems to care about me as a person.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q10",
          "name": "I would describe my workplace as being psychologically healthy.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q11",
          "name": "My department or agency does a good job of raising awareness of mental health in the workplace.",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            4,
            5,
          ],
          "id": "Q12",
          "name": "Overall, my level of work-related stress isâ€¦",
          "negative": Array [
            4,
            5,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            1,
            2,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
            2,
          ],
          "id": "Q13",
          "name": "After my workday, I feel emotionally drained.",
          "negative": Array [
            1,
            2,
          ],
          "neutral": Array [
            3,
          ],
          "positive": Array [
            4,
            5,
          ],
          "type": "numeric",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q14",
          "name": "Having carefully read the definition of harassment, have you been the victim of harassment on the job in the past two years?",
          "negative": Array [
            1,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            2,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q15a",
          "name": "From whom did you experience harassment on the job? Co-workers",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q15b",
          "name": "From whom did you experience harassment on the job? Individuals with authority over me",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q15c",
          "name": "From whom did you experience harassment on the job? Individuals working for me",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q15d",
          "name": "From whom did you experience harassment on the job? Individuals for whom I have a custodial responsibility (e.g., inmates, offenders, patients, detainees)",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q15e",
          "name": "From whom did you experience harassment on the job? Individuals from other departments or agencies",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q15f",
          "name": "From whom did you experience harassment on the job? Members of the public (individuals or organizations)",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q15g",
          "name": "From whom did you experience harassment on the job? Other",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q16",
          "name": "Having carefully read the definition of discrimination, have you been the victim of discrimination on the job in the past two years?               ",
          "negative": Array [
            1,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            2,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q17a",
          "name": "From whom did you experience discrimination on the job? Co-workers",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q17b",
          "name": "From whom did you experience discrimination on the job? Individuals with authority over me",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q17c",
          "name": "From whom did you experience discrimination on the job? Individuals working for me",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q17d",
          "name": "From whom did you experience discrimination on the job? Individuals for whom I have a custodial responsibility (e.g., inmates, offenders, patients, detainees)",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q17e",
          "name": "From whom did you experience discrimination on the job? Individuals from other departments or agencies",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q17f",
          "name": "From whom did you experience discrimination on the job? Members of the public (individuals or organizations)",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
        Object {
          "agree": Array [
            1,
          ],
          "id": "Q17g",
          "name": "From whom did you experience discrimination on the job? Other",
          "negative": Array [
            null,
          ],
          "neutral": Array [
            null,
          ],
          "positive": Array [
            null,
          ],
          "type": "boolean",
        },
      ],
    },
  },
}
`;

exports[`PSES data has survey data for an org 1`] = `
Object {
  "data": Object {
    "root": Object {
      "org": Object {
        "pses_data": Array [
          Object {
            "agree": 81,
            "answer1": 43,
            "answer2": 35,
            "answer3": 10,
            "answer4": 5,
            "answer5": 3,
            "answer6": 3,
            "answer7": 0,
            "answer_count": 129391,
            "average_1_to_5": 4.14,
            "average_percent": 79,
            "negative": 9,
            "neutral": 11,
            "positive": 81,
            "question": Object {
              "id": "Q01",
            },
            "year": "2017",
          },
          Object {
            "agree": 75,
            "answer1": 37,
            "answer2": 38,
            "answer3": 9,
            "answer4": 9,
            "answer5": 6,
            "answer6": 0,
            "answer7": 0,
            "answer_count": 129170,
            "average_1_to_5": 3.89,
            "average_percent": 72,
            "negative": 16,
            "neutral": 9,
            "positive": 75,
            "question": Object {
              "id": "Q02",
            },
            "year": "2017",
          },
          Object {
            "agree": 66,
            "answer1": 31,
            "answer2": 35,
            "answer3": 14,
            "answer4": 11,
            "answer5": 8,
            "answer6": 0,
            "answer7": 0,
            "answer_count": 129123,
            "average_1_to_5": 3.69,
            "average_percent": 67,
            "negative": 20,
            "neutral": 14,
            "positive": 66,
            "question": Object {
              "id": "Q03",
            },
            "year": "2017",
          },
          Object {
            "agree": 59,
            "answer1": 22,
            "answer2": 36,
            "answer3": 16,
            "answer4": 14,
            "answer5": 10,
            "answer6": 1,
            "answer7": 0,
            "answer_count": 129149,
            "average_1_to_5": 3.48,
            "average_percent": 62,
            "negative": 24,
            "neutral": 16,
            "positive": 59,
            "question": Object {
              "id": "Q04",
            },
            "year": "2017",
          },
          Object {
            "agree": 72,
            "answer1": 37,
            "answer2": 34,
            "answer3": 12,
            "answer4": 9,
            "answer5": 7,
            "answer6": 0,
            "answer7": 1,
            "answer_count": 129247,
            "average_1_to_5": 3.85,
            "average_percent": 71,
            "negative": 17,
            "neutral": 12,
            "positive": 72,
            "question": Object {
              "id": "Q05",
            },
            "year": "2017",
          },
          Object {
            "agree": 65,
            "answer1": 28,
            "answer2": 37,
            "answer3": 14,
            "answer4": 12,
            "answer5": 9,
            "answer6": 0,
            "answer7": 0,
            "answer_count": 129339,
            "average_1_to_5": 3.62,
            "average_percent": 66,
            "negative": 21,
            "neutral": 14,
            "positive": 65,
            "question": Object {
              "id": "Q06",
            },
            "year": "2017",
          },
          Object {
            "agree": 78,
            "answer1": 38,
            "answer2": 40,
            "answer3": 10,
            "answer4": 7,
            "answer5": 4,
            "answer6": 0,
            "answer7": 0,
            "answer_count": 129381,
            "average_1_to_5": 4.02,
            "average_percent": 75,
            "negative": 11,
            "neutral": 10,
            "positive": 78,
            "question": Object {
              "id": "Q07",
            },
            "year": "2017",
          },
          Object {
            "agree": 79,
            "answer1": 52,
            "answer2": 27,
            "answer3": 8,
            "answer4": 7,
            "answer5": 6,
            "answer6": 0,
            "answer7": 0,
            "answer_count": 129496,
            "average_1_to_5": 4.12,
            "average_percent": 78,
            "negative": 13,
            "neutral": 8,
            "positive": 79,
            "question": Object {
              "id": "Q08",
            },
            "year": "2017",
          },
          Object {
            "agree": 79,
            "answer1": 53,
            "answer2": 26,
            "answer3": 10,
            "answer4": 6,
            "answer5": 6,
            "answer6": 1,
            "answer7": 0,
            "answer_count": 129278,
            "average_1_to_5": 4.15,
            "average_percent": 79,
            "negative": 11,
            "neutral": 10,
            "positive": 79,
            "question": Object {
              "id": "Q09",
            },
            "year": "2017",
          },
          Object {
            "agree": 60,
            "answer1": 28,
            "answer2": 32,
            "answer3": 13,
            "answer4": 14,
            "answer5": 13,
            "answer6": 0,
            "answer7": 0,
            "answer_count": 129340,
            "average_1_to_5": 3.48,
            "average_percent": 62,
            "negative": 27,
            "neutral": 13,
            "positive": 60,
            "question": Object {
              "id": "Q10",
            },
            "year": "2017",
          },
          Object {
            "agree": 65,
            "answer1": 28,
            "answer2": 35,
            "answer3": 18,
            "answer4": 10,
            "answer5": 7,
            "answer6": 1,
            "answer7": 0,
            "answer_count": 129187,
            "average_1_to_5": 3.7,
            "average_percent": 68,
            "negative": 17,
            "neutral": 18,
            "positive": 65,
            "question": Object {
              "id": "Q11",
            },
            "year": "2017",
          },
          Object {
            "agree": 34,
            "answer1": 5,
            "answer2": 19,
            "answer3": 42,
            "answer4": 22,
            "answer5": 12,
            "answer6": 0,
            "answer7": 0,
            "answer_count": 128015,
            "average_1_to_5": 2.83,
            "average_percent": 46,
            "negative": 34,
            "neutral": 42,
            "positive": 24,
            "question": Object {
              "id": "Q12",
            },
            "year": "2017",
          },
          Object {
            "agree": 33,
            "answer1": 10,
            "answer2": 23,
            "answer3": 40,
            "answer4": 19,
            "answer5": 7,
            "answer6": 0,
            "answer7": 0,
            "answer_count": 127955,
            "average_1_to_5": 2.9,
            "average_percent": 48,
            "negative": 33,
            "neutral": 41,
            "positive": 26,
            "question": Object {
              "id": "Q13",
            },
            "year": "2017",
          },
          Object {
            "agree": 22,
            "answer1": 22,
            "answer2": 78,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 128162,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": 22,
            "neutral": null,
            "positive": 78,
            "question": Object {
              "id": "Q14",
            },
            "year": "2017",
          },
          Object {
            "agree": 50,
            "answer1": 50,
            "answer2": 50,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 26543,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q15a",
            },
            "year": "2017",
          },
          Object {
            "agree": 64,
            "answer1": 64,
            "answer2": 36,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 26545,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q15b",
            },
            "year": "2017",
          },
          Object {
            "agree": 7,
            "answer1": 7,
            "answer2": 93,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 26542,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q15c",
            },
            "year": "2017",
          },
          Object {
            "agree": 5,
            "answer1": 5,
            "answer2": 95,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 26541,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q15d",
            },
            "year": "2017",
          },
          Object {
            "agree": 6,
            "answer1": 6,
            "answer2": 94,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 26540,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q15e",
            },
            "year": "2017",
          },
          Object {
            "agree": 10,
            "answer1": 10,
            "answer2": 90,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 26541,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q15f",
            },
            "year": "2017",
          },
          Object {
            "agree": 4,
            "answer1": 4,
            "answer2": 96,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 26538,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q15g",
            },
            "year": "2017",
          },
          Object {
            "agree": 12,
            "answer1": 12,
            "answer2": 88,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 128926,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": 12,
            "neutral": null,
            "positive": 88,
            "question": Object {
              "id": "Q16",
            },
            "year": "2017",
          },
          Object {
            "agree": 36,
            "answer1": 36,
            "answer2": 64,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 14972,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q17a",
            },
            "year": "2017",
          },
          Object {
            "agree": 79,
            "answer1": 79,
            "answer2": 21,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 14975,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q17b",
            },
            "year": "2017",
          },
          Object {
            "agree": 4,
            "answer1": 4,
            "answer2": 96,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 14972,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q17c",
            },
            "year": "2017",
          },
          Object {
            "agree": 4,
            "answer1": 4,
            "answer2": 96,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 14970,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q17d",
            },
            "year": "2017",
          },
          Object {
            "agree": 8,
            "answer1": 8,
            "answer2": 92,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 14972,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q17e",
            },
            "year": "2017",
          },
          Object {
            "agree": 9,
            "answer1": 9,
            "answer2": 91,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 14972,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q17f",
            },
            "year": "2017",
          },
          Object {
            "agree": 7,
            "answer1": 7,
            "answer2": 93,
            "answer3": null,
            "answer4": null,
            "answer5": null,
            "answer6": null,
            "answer7": null,
            "answer_count": 14972,
            "average_1_to_5": null,
            "average_percent": null,
            "negative": null,
            "neutral": null,
            "positive": null,
            "question": Object {
              "id": "Q17g",
            },
            "year": "2017",
          },
        ],
      },
    },
  },
}
`;
